### PR TITLE
[#14] Integrate Home drawer

### PR DESCRIPTION
**Pull request by chornerman**
#14

## What happened 👀

- [x] Display the user name and user profile image at the top of the drawer
- [x] When tapping on "Logout", we will call Logout API through the view model
  - We will close the drawer and show Loading Indicator
  - If the API call return Success, we will show the Login screen
  - If the API call return Failed, we will show the SnackBar with the error message
- [x] Display the app version at the bottom of the drawer
- [x] Add unit tests for `HomeViewModel`
- [x] Refactor `Navigator`

## Insight 📝

- We use [package_info_plus](https://pub.dev/packages/package_info_plus) for retrieving the app version

## Proof Of Work 📹

https://user-images.githubusercontent.com/13492460/209561732-903ea5dd-2f13-41a2-b699-264aeca79041.mov

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -28,7 +27,6 @@ class App extends StatelessWidget {
       theme: AppTheme.defaultTheme,
       localizationsDelegates: AppLocalizations.localizationsDelegates,
       supportedLocales: AppLocalizations.supportedLocales,
-      home: LoginPage(),
```
Comment (luongvo) [inline lib/main.dart:31]: 👍

File: lib/navigator.dart [hunk 2]
```diff
@@ -36,4 +38,8 @@ class AppNavigatorImpl extends AppNavigator {
   @override
   void navigateToResetPassword(BuildContext context) =>
       Navigator.of(context).pushNamed(_routeResetPassword);
+
+  @override
+  void navigateToLoginAndClearStack(BuildContext context) =>
+      Navigator.pushNamedAndRemoveUntil(context, _routeLogin, (r) => false);
```
Comment (luongvo) [inline lib/navigator.dart:44]: 👍

File: lib/page/home/home_page.dart [hunk 3]
```diff
@@ -57,38 +65,51 @@ class _HomePageState extends ConsumerState<HomePage> {
 
   @override
   Widget build(BuildContext context) {
+    ref.listen<HomeState>(homeViewModelProvider, (
+      HomeState? previousState,
```
Comment (luongvo) [inline lib/page/home/home_page.dart:69]: same, `previousState` has no usage, use `_`

**Review by luongvo — APPROVED** [review]
minor issue, lgtm

File: lib/page/home/home_page.dart (hunk 3 above)
Comment (chornerman) [reply lib/page/home/home_page.dart:69]: Fixed in https://github.com/chornerman/Survey/pull/108

**Review by markgravity — APPROVED** [review]

**Pull request merged**